src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -152,6 +152,10 @@ private BoundExpression CompareWithConstant(BoundExpression input, BoundExpressi
                     return _factory.ObjectEqual(_factory.Convert(systemObject, input), boundConstant);
                 }
             }
+            else if (input.Type.IsNullableType() && boundConstant is BoundDefaultExpression boundDefault && boundDefault.NullableNeverHasValue())
```

Comment:
It is likely that, according to the language, expressions on nullable type cannot have constant values.